app/api/gql/schema/resolver-draft.go

```diff
@@ -0,0 +1,25 @@
+package schema
+
+import (
+	"time"
+)
+
+type Task struct {
+	// this is always injected by a parent level resolver which returns a Task, at runtime
+	deps           DependencyObject
```

Comment:
```suggestion
	deps           Dependencies
```

Great! I think this will allow the business logic to evolve without constantly adding or removing dependencies.